### PR TITLE
fix #294  #274 EnableCircuitBreaker error & add nacos gateway example

### LINKED ISSUE #294: Is the better to enable sentinel with EnableCircuitBreaker?
I found spring-cloud-alibaba-sentinel 's spring.factories config 
```
org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
org.springframework.cloud.alibaba.sentinel.SentinelWebAutoConfiguration,\
org.springframework.cloud.alibaba.sentinel.endpoint.SentinelEndpointAutoConfiguration,\
org.springframework.cloud.alibaba.sentinel.custom.SentinelAutoConfiguration,\
org.springframework.cloud.alibaba.sentinel.feign.SentinelFeignAutoConfiguration
```
Is this the better replace "@EnableAutoConfiguration" to "@EnableCircuitBreaker"?

### PR DESCRIPTION
**Pull request by pig4cloud**
This is done to solve the problem of using standard Spring Cloud annotations.

1 . import spring-cloud-commons
2. new blank Configuration SentinelCircuitBreakerConfiguration
3. set up spring.factories EnableCircuitBreaker
As the isuue say,@EnableCircuitBreaker annotation doesn't really do anything, it's just an identifier.

### CODE REVIEW

File: spring-cloud-alibaba-sentinel/pom.xml [hunk 1]
```diff
@@ -42,6 +42,12 @@
             <optional>true</optional>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-commons</artifactId>
+            <optional>true</optional>
+        </dependency>
+
```
Comment (fangjian0423) [inline spring-cloud-alibaba-sentinel/pom.xml]: Maybe the scope of this dependency should be provided and using @ConditionalOnClass to annotate SentinelCircuitBreakerConfiguration.

**Comment by fangjian0423** [comment]
LGTM

**Comment by pig4cloud** [comment]
thx

**Comment by flystar32** [comment]
Thank you for your contribution to Spring Cloud Alibaba!

**Pull request merged**